__tests__/integration/plugins/plugin-events.test.js

```diff
@@ -0,0 +1,92 @@
+import { TestUtils } from '../utils/test-utils-integration';
+import { WalletHelper } from '../utils/wallet-helper';
+import { notificationBus } from '../../../src/services/notification.service';
+import * as childManager from '../../../src/plugins/child';
+
+let loadedPlugin = null;
+const pluginWalletId = 'plugin-wallet';
+
+describe('test the plugin event handler', () => {
+  /** @type {WalletHelper} */
+  let wallet1;
+
+  beforeAll(async () => {
+    /*
+     * Mimic the behavior of `src/plugins/child.js` because it is not called within the context of
+     * Jest. However, here we initialize the plugin within the main process to allow for integration
+     * testing.
+     *
+     * @see https://github.com/HathorNetwork/hathor-wallet-headless/blob/master/src/index.js
+     * @see https://github.com/HathorNetwork/hathor-wallet-headless/blob/master/src/plugins/child.js
+     */
+    childManager.hathorPlugins.integration = {
+      name: 'integration',
+      file: '../../__tests__/integration/plugins/integration_test_plugin',
+    };
+    [loadedPlugin] = await childManager.loadPlugins(['integration'], {});
+    loadedPlugin.init(notificationBus);
+
+    try {
+      // Initialize an empty wallet
+      wallet1 = WalletHelper.getPrecalculatedWallet(pluginWalletId);
+      await WalletHelper.startMultipleWalletsForTest([wallet1]);
+    } catch (err) {
+      TestUtils.logError(err.stack);
+    }
+  });
+
+  afterAll(async () => {
+    await wallet1.stop();
+    loadedPlugin.close();
+  });
+
+  it('should return the events that led to complete wallet start', async () => {
+    const currentHistory = loadedPlugin
+      .retrieveEventHistory();
+
+    // The full event history should have more elements than those of the test wallet
+    const testWalletHistory = currentHistory.filter(event => event.walletId === pluginWalletId);
+    expect(currentHistory.length).toBeGreaterThan(testWalletHistory.length);
+
+    // There must have been 5 wallet changes for this test, one with each step of its initialization
+    expect(testWalletHistory.filter(event => event.type === 'wallet:state-change')).toHaveLength(5);
```

Comment:
suggestion(if-minor): test that we have `wallet:state-change` instead of the number of state changes.

This is just to avoid future changes on the lib to have an effect on these tests.

Reply:
Good point. Done on [182dbec](https://github.com/HathorNetwork/hathor-wallet-headless/pull/407/commits/182dbec7a0d4f11cf811a29b5e2f7dc3edf33a02).